front/app/components/admin/Pagination/index.tsx

```diff
@@ -1,14 +1,31 @@
 import React from 'react';
+import { useLocation } from 'react-router-dom';
 import { Menu } from 'semantic-ui-react';
+import clHistory from 'utils/cl-router/history';
+import { stringify, parse } from 'qs';
 
 // Typing
 interface Props {
   currentPage: number;
   totalPages: number;
-  loadPage: (arg: number) => void;
+  loadPage?: (arg: number) => void;
 }
 
 const Pagination = ({ currentPage, totalPages, loadPage }: Props) => {
+  const location = useLocation();
+  const query = parse(location.search, { ignoreQueryPrefix: true });
+
+  const handlePaginationClick = (newPageNumber: number) => {
+    clHistory.push({
+      pathname: location.pathname,
+      search: stringify(
+        { ...query, pageNumber: newPageNumber },
+        { addQueryPrefix: true }
+      ),
+    });
+  };
+  const handleLoadPage = loadPage || handlePaginationClick;
```

Comment:
Hmm, not sure if we should add this code here...

I saw you removed the handlers from CustomEmails/All, but isn't this just an query param update for useCampaigns that we should handle in that code? All other places that use Pagination specify this prop, so let's try to stick to that?

Reply:
i added pagination with react query i couldn't find an example so i hope is ok. it works though